### ISSUE TITLE
Meaning of the attributions' sign in integrated gradients

### ISSUE THREAD

**Issue by kekehia123** [pytorch/captum#685]
## ❓ Questions and Help
Hi, 

I'm confused about whether we should care about the sign of the attributions output by integrated gradients method? Do large negative values also mean the feature is important for classification of the specific category?

Thanks a lot!

Xinke Shen

**Comment by bilalsal** [comment]
Hi Xinke,

apologies for responding late. Yes, large negative values also reflect that a feature has a strong influence on the output. The sign indicates whether the scalar target you specified will be (generally) lower if the feature increases. Note that Captum does not yet account for feature interaction, and hence non-linear dependencies between outputs and inputs are not captured by the computed attribution scores.

Hope this helps

**Comment by kekehia123** [comment]
> Hi Xinke,
> 
> apologies for responding late. Yes, large negative values also reflect that a feature has a strong influence on the output. The sign indicates whether the scalar target you specified will be (generally) lower if the feature increases. Note that Captum does not yet account for feature interaction, and hence non-linear dependencies between outputs and inputs are not captured by the computed attribution scores.
> 
> Hope this helps

Thank you very much! Your response helps a lot!

**Comment by bilalsal** [comment]
Hi @kekehia123,

Yes it is quite possible for the same feature to vote for class 0 when present in certain samples, and for class 1 when present in other samples.

Sometimes, this means the feature might be not very discriminative, and is not the main reason for the classification output. To determine if this is really the case I recommend you use [FeatureAblation](https://captum.ai/api/feature_ablation.html) and [FeaturePermutation](https://captum.ai/api/feature_permutation.html).

Other times, this might indicate feature interaction: The feature you mention only contributes to class 1 if another feature (or group of features) is within specific value ranges. Currently, Captum does not explicitly account for such feature interaction.
More on feature interaction: https://arxiv.org/abs/2006.10965 (also check out the Archipelago library by Michael Tsang and others https://github.com/mtsang/archipelago).

Hope this helps

**Comment by kekehia123** [comment]
> Hi @kekehia123,
> 
> Yes it is quite possible for the same feature to vote for class 0 when present in certain samples, and for class 1 when present in other samples.
> 
> Sometimes, this means the feature might be not very discriminative, and is not the main reason for the classification output. To determine if this is really the case I recommend you use [FeatureAblation](https://captum.ai/api/feature_ablation.html) and [FeaturePermutation](https://captum.ai/api/feature_permutation.html).
> 
> Other times, this might indicate feature interaction: The feature you mention only contributes to class 1 if another feature (or group of features) is within specific value ranges. Currently, Captum does not explicitly account for such feature interaction.
> More on feature interaction: https://arxiv.org/abs/2006.10965 (also check out the Archipelago library by Michael Tsang and others https://github.com/mtsang/archipelago).
> 
> Hope this helps

Thanks a lot! Your answer helps a lot!

I found there should be a bug in my code: my inputs have values from around -2 to 2, so I guess the baseline should not be 0s? I think the baseline should be smaller than all the input values, or the sign of the attributions are definitely inconsistent? I set the baseline to -5 and the results seem reasonable.

However, I have two more questions now... Sorry for bothering you again.

As I did binary classification, I can directly conduct a two-sample t-test on each feature dimension, and I compare the t values to the attributions of the integrated gradients (IG) algorithm. I found some of the most important features identified by IG algorithm do not seem to be significantly different between the two classes. Do you have any explanations for this? I think IG is looking for the linear influence of individual feature dimensions on the output. It does not make sense the important feature is not different between different classes? 

The second question is about feature permutation. According to the pseudocode for the [FeaturePermutation](https://captum.ai/api/feature_permutation.html), the model needs to receive the "batch_labels" to calculate the error, but I don't know how to pass the argument to the function? The example seems not to pass the labels. Could you help me with it? 

Thank you very much!

**Comment by bilalsal** [comment]
Hi @kekehia123,

your idea of using min(x) as  baseline is interesting. It might indeed better fit the nature of your data and classification task.
A zero baseline could still fit (even if the values can range from -2 to 2), it really depends on the data and the task.
This Distill article sheds light into how the choice of baselines affect the attribution and provides some recommendations on that: https://distill.pub/2020/attribution-baselines/
@vivekmig , @NarineK: Any comments on @kekehia123's idea to use min(x) as a baseline?

I do not think the t-tests on individual features found important by IG help us make conclusions about these features. As I tried to outline in my previous reply, the importance of a feature for a certain output might change between different samples: Therefore, you might not see a consistent pattern across all samples, and t-tests will not help establish any such pattern.
More importantly, your features likely interact in non-linear ways as they get used by the model to make certain predictions.

Accordingly, a single feature will hardly be discriminative (t-test wise). If it were, you probably do not need a deep neural network to solve your problem (a naive-bayesian classifier or a simple logistic classifier would have been sufficient).

Regarding FeaturePermutation, Captum does not really need labels as in the illustrative pseudocode. Instead, Captum will treat the `target` you provide (e.g. `target=1` for class 1) to measure the impact of permuting each feature on this target. If the impact is minimal, the feature is likely not important for the classifier. If the impact is significant, the feature is important.

Hope this helps
